0.10.3 re-seals if it cant revoke DB creds

**Describe the bug**

```
2018-06-28T03:00:03.448Z [ERROR] expiration: failed to revoke lease: lease_id=database/creds/rds-staging/c273036b-552f-9977-e17a-c552f83f3b2c error="failed to revoke entry: resp: (*logical.Response)(nil) err: error during revoke: could not find role with name "rds-staging" or embedded revocation db name data"                                   
2018-06-28T03:00:03.448Z [ERROR] expiration: failed to revoke lease: lease_id=database/creds/rds-staging/a7d43492-1646-e0b9-2621-9390666d81ea error="failed to revoke entry: resp: (*logical.Response)(nil) err: error during revoke: could not find role with name "rds-staging" or embedded revocation db name data"                                                                            
2018-06-28T03:00:03.449Z [ERROR] expiration: failed to revoke lease: lease_id=database/creds/rds-staging/c0e9bd84-44b3-d3a5-aa5b-d3318ddb0a97 error="failed to revoke entry: resp: (*logical.Response)(nil) err: error during revoke: could not find role with name "rds-staging" or embedded revocation db name data"                                  
panic: interface conversion: interface {} is []interface {}, not []string                                                                                                                                                                                                                                                     
goroutine 1305 [running]:                                                                                                                                                                                                                     github.com/hashicorp/vault/builtin/logical/database.(*databaseBackend).secretCredsRevoke.func1(0x2586340, 0xc420de3f40, 0xc4219217a0, 0xc420770e30, 0x0, 0x0, 0x0)
        /gopath/src/github.com/hashicorp/vault/builtin/logical/database/secret_creds.go:114 +0x7e8                                                                                                                                            github.com/hashicorp/vault/logical/framework.(*Secret).HandleRevoke(0xc420d84870, 0x2586340, 0xc420de3f40, 0xc4219217a0, 0xc4219214f8, 0x10094709b11fb01, 0x0)
        /gopath/src/github.com/hashicorp/vault/logical/framework/secret.go:87 +0x9a                                                                                                                                                           github.com/hashicorp/vault/logical/framework.(*Backend).handleRevokeRenew(0xc420972a90, 0x2586340, 0xc420de3f40, 0xc4219217a0, 0x64363966c156b618, 0x6664612d37613835, 0xc421495988)
        /gopath/src/github.com/hashicorp/vault/logical/framework/backend.go:394 +0x1fb                                                                                                                                                        github.com/hashicorp/vault/logical/framework.(*Backend).HandleRequest(0xc420972a90, 0x2586340, 0xc420de3f40, 0xc4219217a0, 0x0, 0x0, 0x0)
        /gopath/src/github.com/hashicorp/vault/logical/framework/backend.go:169 +0x6fb                                                                                                                                                        github.com/hashicorp/vault/vault.(*Router).routeCommon(0xc4208e0f00, 0x2586340, 0xc420de3f40, 0xc4219217a0, 0x410600, 0x0, 0x22e0000, 0x0, 0x0)
        /gopath/src/github.com/hashicorp/vault/vault/router.go:523 +0x7c2                                                                                                                                                                     github.com/hashicorp/vault/vault.(*Router).Route(0xc4208e0f00, 0x2586340, 0xc420de3f40, 0xc4219217a0, 0x0, 0xc421495dc0, 0x19de3e1)
        /gopath/src/github.com/hashicorp/vault/vault/router.go:378 +0x4e                                                                                                                                                                      github.com/hashicorp/vault/vault.(*ExpirationManager).revokeEntry(0xc420621200, 0xc4213550e0, 0x3f, 0xc4213550e0)
        /gopath/src/github.com/hashicorp/vault/vault/expiration.go:1092 +0x1c7                                                                                                                                                                github.com/hashicorp/vault/vault.(*ExpirationManager).revokeCommon(0xc420621200, 0xc42011b440, 0x3f, 0x0, 0x0, 0x0)                     
        /gopath/src/github.com/hashicorp/vault/vault/expiration.go:510 +0x491                                                                                                                                                                 github.com/hashicorp/vault/vault.(*ExpirationManager).Revoke(0xc420621200, 0xc42011b440, 0x3f, 0x0, 0x0) 
        /gopath/src/github.com/hashicorp/vault/vault/expiration.go:489 +0x112                                                                                                                                                                 github.com/hashicorp/vault/vault.(*ExpirationManager).expireID(0xc420621200, 0xc42011b440, 0x3f)          
        /gopath/src/github.com/hashicorp/vault/vault/expiration.go:1066 +0x1ce                                                                                                                                                                github.com/hashicorp/vault/vault.(*ExpirationManager).updatePendingInternal.func1()                        
        /gopath/src/github.com/hashicorp/vault/vault/expiration.go:1031 +0x3f                                                                                                                                                                 created by time.goFunc                                                                                                                              
        /goroot/src/time/sleep.go:172 +0x44                                                                                                                                                                                                   ==> Vault server configuration:    
```

**To Reproduce**
Steps to reproduce the behavior:
1. create temp db creds (psql in this case) via database backend
2. Tear down the database

**Expected behavior**
in 0.9.6 nothing happened.
In 0.10.3 the server bounces and re-seals


**Environment:**
* 0.10.3

Vault server configuration file(s):

n/a

**Additional context**

We tear down staging nightly. During the day we let devs create temp creds using the database provider

--- a/builtin/logical/database/backend_test.go
+++ b/builtin/logical/database/backend_test.go
@@ -603,7 +603,7 @@ func TestBackend_basic(t *testing.T) {
 					"username":              credsResp.Data["username"],
 					"role":                  "plugin-role-test",
 					"db_name":               "plugin-test",
-					"revocation_statements": []string(nil),
+					"revocation_statements": []interface{}(nil),
 				},
 			},
 		})

--- a/builtin/logical/database/secret_creds.go
+++ b/builtin/logical/database/secret_creds.go
@@ -111,7 +111,9 @@ func (b *databaseBackend) secretCredsRevoke() framework.OperationFunc {
 			if statementsRaw, ok := req.Secret.InternalData["revocation_statements"]; !ok {
 				return nil, fmt.Errorf("error during revoke: could not find role with name %q or embedded revocation statement data", req.Secret.InternalData["role"])
 			} else {
-				statements.Revocation = statementsRaw.([]string)
+				for _, v := range statementsRaw.([]interface{}) {
+					statements.Revocation = append(statements.Revocation, v.(string))
+				}
 			}
 		}
 

